app/assets/javascripts/discourse/controllers/move-to-topic.js.es6

```diff
@@ -0,0 +1,217 @@
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import { movePosts, mergeTopic } from "discourse/models/topic";
+import DiscourseURL from "discourse/lib/url";
+import { default as computed } from "ember-addons/ember-computed-decorators";
+import { extractError } from "discourse/lib/ajax-error";
+
+export default Ember.Controller.extend(ModalFunctionality, {
+  topicName: null,
+  saving: false,
+  categoryId: null,
+  tags: null,
+  canAddTags: Ember.computed.alias("site.can_create_tag"),
+  canTagMessages: Ember.computed.alias("site.can_tag_pms"),
+  selectedTopicId: null,
+  newTopic: Ember.computed.equal("selection", "new_topic"),
+  existingTopic: Ember.computed.equal("selection", "existing_topic"),
+  newMessage: Ember.computed.equal("selection", "new_message"),
+  existingMessage: Ember.computed.equal("selection", "existing_message"),
+  participants: null,
+
+  topicController: Ember.inject.controller("topic"),
+  selectedPostsCount: Ember.computed.alias(
+    "topicController.selectedPostsCount"
+  ),
+  selectedAllPosts: Ember.computed.alias("topicController.selectedAllPosts"),
+  selectedPosts: Ember.computed.alias("topicController.selectedPosts"),
+
+  @computed("saving", "selectedTopicId", "topicName")
+  buttonDisabled(saving, selectedTopicId, topicName) {
+    return (
+      saving || (Ember.isEmpty(selectedTopicId) && Ember.isEmpty(topicName))
+    );
+  },
+
+  @computed(
+    "saving",
+    "newTopic",
+    "existingTopic",
+    "newMessage",
+    "existingMessage"
+  )
+  buttonTitle(saving, newTopic, existingTopic, newMessage, existingMessage) {
+    if (newTopic) {
+      return I18n.t("topic.split_topic.title");
+    } else if (existingTopic) {
+      return I18n.t("topic.merge_topic.title");
+    } else if (newMessage) {
+      return I18n.t("topic.move_to_new_message.title");
+    } else if (existingMessage) {
+      return I18n.t("topic.move_to_existing_message.title");
+    } else {
+      return I18n.t("saving");
+    }
+  },
+
+  onShow() {
+    this.setProperties({
+      "modal.modalClass": "move-to-modal",
+      saving: false,
+      selection: "new_topic",
+      categoryId: null,
+      topicName: "",
+      tags: null,
+      participants: null
+    });
+
+    const isPrivateMessage = this.get("model.isPrivateMessage");
+    const canSplitTopic = this.get("canSplitTopic");
+    if (isPrivateMessage) {
+      if (canSplitTopic) {
```

Comment:
This would be a good candidate for the ternary operator since most of the line is the same.